src/ray/core_worker/object_interface.cc

```diff
@@ -1,25 +1,142 @@
 #include "object_interface.h"
+#include "context.h"
+#include "core_worker.h"
 
 namespace ray {
 
-Status CoreWorkerObjectInterface::Put(const Buffer &buffer, const ObjectID *object_id) {
+const int CoreWorkerObjectInterface::c_get_timeout_ms_ = 1000;
+
+/// Represents a byte buffer for plasma object.
+class PlasmaBuffer : public Buffer {
+ public:
+  PlasmaBuffer(std::shared_ptr<arrow::Buffer> buffer) : buffer_(buffer) {}
+
+  uint8_t *Data() const override { return const_cast<uint8_t*>(buffer_->data()); }
+
+  size_t Size() const override { return buffer_->size(); }
+
+ private:
+  /// shared_ptr to arrow buffer which can potentially hold a reference
+  /// for the object (when it's a plasma::PlasmaBuffer).
+  std::shared_ptr<arrow::Buffer> buffer_;
+};
+
+CoreWorkerObjectInterface::CoreWorkerObjectInterface(CoreWorker &core_worker)
+  : core_worker_(core_worker) {
+  RAY_ARROW_CHECK_OK(store_client_.Connect(core_worker_.store_socket_));
+}
+
+Status CoreWorkerObjectInterface::Put(const Buffer &buffer, ObjectID *object_id) {
+  auto &context = core_worker_.GetContext();
+  ObjectID put_id = ObjectID::for_put(context.GetCurrentTaskID(),
+      context.GetNextPutIndex());
+  *object_id = put_id;
+
+  auto plasma_id = put_id.to_plasma_id();
+  std::shared_ptr<arrow::Buffer> data;
+  auto status = store_client_.Create(plasma_id, buffer.Size(), nullptr, 0, &data);
+  if (!status.ok()) {
+    return ray::Status::IOError(status.message());
+  }
+  memcpy(data->mutable_data(), buffer.Data(), buffer.Size());
+  RAY_ARROW_CHECK_OK(store_client_.Seal(plasma_id));
+  RAY_ARROW_CHECK_OK(store_client_.Release(plasma_id));
```

Comment:
We should return error status, instead of failing. So we can propagate error to Java/Python. 
(Also apply to other places.)

Reply:
ok, removed all these checks except the `Connect()` in constructor.

Reply:
We should still check status, but do not crash the process. Maybe we can add a `RETURN_IF_NOT_OK` macro that does this:
```
if (!status.ok()) { return status; }
```

Reply:
that would be better, updated.